ArduPlane/geofence.cpp

```diff
@@ -291,14 +291,14 @@ void Plane::geofence_check(bool altitude_check_only)
         // GUIDED to the return point
         if (geofence_state != nullptr &&
             (g.fence_action == FENCE_ACTION_GUIDED || g.fence_action == FENCE_ACTION_GUIDED_THR_PASS || g.fence_action == FENCE_ACTION_RTL) &&
-            (control_mode == GUIDED || control_mode == AVOID_ADSB) &&
+            (control_mode == &mode_guided || control_mode == &mode_avoidADSB) &&
             geofence_present() &&
             geofence_state->boundary_uptodate &&
             geofence_state->old_switch_position == oldSwitchPosition &&
             guided_WP_loc.lat == geofence_state->guided_lat &&
             guided_WP_loc.lng == geofence_state->guided_lng) {
             geofence_state->old_switch_position = 254;
-            set_mode(get_previous_mode(), MODE_REASON_GCS_COMMAND);
+            set_mode(*previous_mode, MODE_REASON_GCS_COMMAND);
```

Comment:
Some sanity checks here for `previous_mode` not being `nullptr`.  Best to be safe.

Reply:
I've added this check, although I think previous_mode will get removed. Hopefully removed. It's kind of a hack the way it is now.